docker/docker.md

```diff
@@ -0,0 +1,87 @@
+# sim-ln Docker Setup
+
+This guide provides instructions on how to build, run, and manage the `sim-ln` Docker container using the provided Makefile.
+
+## Prerequisites
+
+- Docker installed on your machine.
+- A `config.json` file that contains your specific configuration.
+
+## Building the Docker Image
+
+To build the Docker image, execute:
+
+```bash
+make build-docker
+```
+
+This command will make the necessary preparations and build the Docker image.
+
+## Mounting the Volume
+
+To ensure that the Docker container can access the configuration and other necessary data, we use Docker volumes. Before you can run the container, you need to set up the volume with the necessary data.
+
+To do this:
+
+```bash
+make mount-volume CONFIG_PATH=/path/to/your/config.json
+```
+
+Replace `/path/to/your/config.json` with the actual path to your `config.json` file.
+
+*Note:* The path to your config must be an absolute path not relative e.g. "/Users/anonymous/bitcoin-dev-project/sim-ln/config.json"
+
+## Running the Docker Container
+
+To run the `sim-ln` Docker container:
+
+```bash
+make run
+```
+
+You can adjust the logging level by providing the `LOG_LEVEL` variable. The default value is `info`. Example:
+
+```bash
+make run LOG_LEVEL=debug
+```
+
+Other configurable variables include:
+
+- `HELP`: Set to `true` to print the help message.
+- `PRINT_BATCH_SIZE`: Set the batch size for printing the results.
+- `TOTAL_TIME`: Adjusts the total time.
+
+Example usage:
+
+```bash
+make run PRINT_BATCH_SIZE=100 TOTAL_TIME=5000
+```
+
+For an interactive session:
+
+```bash
+make run-interactive
+```
+
+## Stopping the Container
+
+To stop the `sim-ln` Docker container, use:
+
+```bash
+make stop
+```
+
+## Help
+
+For a list of available Makefile commands and their descriptions, run:
+
+```bash
+make help
+```
+
+---
+
+## Notes for Developers:
+
+- When you use the `mount-volume` command, it sets up a Docker volume named `simln-config`. This volume contains your `config.json` file and other necessary files, ensuring that the data is accessible inside the Docker container.
+- The paths to certificates, macaroons, etc., in your `config.json` are automatically adjusted to point to their corresponding paths inside the Docker volume.
```

Comment:
re-emphasize that IP address is not converted while we're here

Reply:
Agreed. I had to go look for the `host.docker.internal` thing to make it run with polar